libevmasm/Assembly.cpp

```diff
@@ -1422,6 +1436,13 @@ LinkerObject const& Assembly::assembleEOF() const
 
 	ret.bytecode += m_auxiliaryData;
 
+	auto appendedDataAndAuxDataSize = ret.bytecode.size() - dataStart;
+
+	// If some data was already added to data section we need to update data section refs accordigly
+	if (appendedDataAndAuxDataSize > 0)
+		for (auto [pos, val] : dataSectionRef)
```

Comment:
```suggestion
		for (auto [refPosition, originalDataOffset]: dataSectionRefs)
```

BTW, since `dataSectionSizeOffset` is the offset from beginning of the header, it's really an absolute value, so perhaps `dataSectionSizePosition` could be a clearer name.

Reply:
Also, I think that both `dataLength` and `appendedDataAndAuxDataSize` will have the same value (you're not appending anything in between) so one of them is redundant.

Reply:
Not any longer. After adding forgotten code fragment it makes more sense now.

Reply:
This should also be renamed to match:
`codeSectionSizeOffsets` -> `codeSectionSizePositions`